xonsh/ptk/shell.py

```diff
@@ -142,8 +142,6 @@ def rprompt_tokens(self, cli):
         prompt.
         """
         p = builtins.__xonsh_env__.get('RIGHT_PROMPT')
-        if len(p) == 0:
-            return []
```

Comment:
I don't think that this check has value because the user function could just as easily return an empty string, in which case we'd call partial_format_prompt on an empty string anyway. For that reason, partial_format_prompt should properly handle empty strings anyway (and I verified that it does: ```list(string.Formatter().parse('')) == []```).